src/nbl/asset/interchange/CSTLMeshFileLoader.cpp

```diff
@@ -253,14 +254,18 @@ SAssetBundle CSTLMeshFileLoader::loadAsset(IReadFile* _file, const IAssetLoader:
 	const size_t vtxSize = hasColor ? (3 * sizeof(float) + 4 + 4) : (3 * sizeof(float) + 4);
 	auto vertexBuf = core::make_smart_refctd_ptr<asset::ICPUBuffer>(vtxSize * positions.size());
 
-	uint32_t normal{};
+	using quant_normal_t = CQuantNormalCache::value_type_t<EF_A2B10G10R10_SNORM_PACK32>;
+
+	quant_normal_t normal;
 	for (size_t i = 0u; i < positions.size(); ++i)
 	{
 		if (i % 3 == 0)
 			normal = quantNormalCache->quantize<EF_A2B10G10R10_SNORM_PACK32>(normals[i / 3]);
 		uint8_t* ptr = ((uint8_t*)(vertexBuf->getPointer())) + i * vtxSize;
 		memcpy(ptr, positions[i].pointer, 3 * 4);
-		((uint32_t*)(ptr + 12))[0] = normal;
+
+		((quant_normal_t*)((uint32_t*)(ptr + 12)))[0] = normal;
```

Comment:
no need for double cast, just do `*reinterpret_cast<quant_normal_t*>(ptr+12) = normal;`

Reply:
`ptr` is `uint8_t*`, wrong pointer arithmetic will be applied

Reply:
here you're doing the exact same arithmetic.

you cast to `uint32_t` after incrementing the pointer by 12 bytes.

Reply:
oh sorry of course

Reply:
also pls dont use c-style casts, it's already unreadable because of amount of parenthesis

Reply:
done